### PR TITLE
Feature/remove path card logic and test

### PR DESCRIPTION
**Pull request by JohnsonMao**
Would it be possible to include events for removing cards in the BoardEvent type as well, so that getCurrentPlacements can obtain the correct PathCard on the board?

Close: #57

### CODE REVIEW

File: apps/backend/src/action/logic/remove-path-card.ts [hunk 1]
```diff
@@ -1,27 +1,108 @@
...
-  //@todo: append event to event source
-  return errAsync(new Error("not implemented"));
-};
+  /**
+   * @question
+   *  Would it be possible to include events for removing cards in the BoardEvent type as well,
+   *  so that getCurrentPlacements can obtain the correct PathCard on the board?
+   */
+  getCurrentPlacements(source as EventSource<BoardEvent>)
+    //@todo: check path card has been placed
```
Comment (kayac-chang) [inline apps/backend/src/action/logic/remove-path-card.ts:103]: @JohnsonMao could you help me clear the todo comment after implementation finished, thx.

File: apps/backend/src/action/logic/remove-path-card.ts [hunk 2]
```diff
@@ -1,27 +1,108 @@
...
+      board,
+      findPlacementByPosition(command),
+      Either.fromOption(() =>
+        TargetCannotBeRemovedError("empty placement cannot be removed")
+      ),
+      Either.chain(isRemovablePathCard),
+      Either.matchW(err, always(ok(command)))
+    );
+
+const appendEventToEventSource =
```
Comment (kayac-chang) [inline apps/backend/src/action/logic/remove-path-card.ts:72]: @JohnsonMao 
To improve the code readability.
Can you help me update it by the following code? 
```ts
    pipe(
      PathCardHasBeenRemovedEvent(command.data),
      (event) =>
        ResultAsync.fromPromise(
          source.append(event),
          always(RepositoryWriteError("failed to write event to repository"))
        )
        .map(() => event)
    );
```

File: apps/backend/src/action/logic/remove-path-card.ts [hunk 3]
```diff
@@ -1,27 +1,108 @@
...
+  TargetCannotBeRemovedError
+>(
+  (placement) => !NonRemovablePathCard.includes(placement.card),
+  (placement) =>
+    TargetCannotBeRemovedError(
+      `the path card ${placement.card} cannot be removed`
+    )
+);
+
+const findPlacementByPosition =
```
Comment (kayac-chang) [inline apps/backend/src/action/logic/remove-path-card.ts:54]: We can omit the second argument in this case.
```ts
const findPlacementByPosition = (command: RockfallCommand) =>
  Array.findFirst<Placement>(
    flow(prop("position"), Vec.eq(command.data.position))
  );
```

**Review by kayac-chang — CHANGES_REQUESTED** [review]

**Comment by kayac-chang** [comment]
@JohnsonMao 
For the branch name, 
I expect it to be `<branch-category>/<branch-name>`, 

The reason is
1. We already know who was working on this PR by commit.
2. The CI will create a preview instance using `<branch-name>.`

**Comment by JohnsonMao** [comment]
Move to PR ( #65 )

**Pull request closed without merging**